### ISSUE TITLE
Unit test compact/table mode

### ISSUE THREAD

**Issue by jeremejevs** [firasdib/Regex101#893]
My usecase is URL matching, where I take an expression like `/^https?:\/\/news\.ycombinator\.com(?!\/(item|user))/` and match it against 5-15 tests cases. I'm aware of the **Expand all tests** button, but the expanded list is pretty hard to parse, especially in the dark mode. This makes it cumbersome to come up with new tests when modifying an expression (caught myself adding duplicate/redundant tests multiple times now).

For me, **Description** and **Target** are wasted space; the best description for all my tests would be `${assertion} ${testString}` in a mono font, aligned by `testString`'s left side, on a single line. Optionally sortable/filterable.

As has been mentioned in #696, there's no one-size-fits-all, so that's why I'm proposing this as a display mode. There should probably be a way to set it as default in user settings too. Thoughts?

**Comment by firasdib** [comment]
I see what you're saying, and I'm open to a revised UI.

**Comment by firasdib** [comment]
Thought about this some more:

I disagree that `description` is pointless. To me its the most valuable. Tests need to have a good description, such as, `Ensure regex does not match X in the wrong position`. Then you (and others) can easily find out what this test is supposed to do. The test string might be complete mumbo jumbo.

If the issue is contrast in dark mode, I'd be happy to address that, if you could give me some more details.

**Comment by jeremejevs** [comment]
@firasdib Thanks for taking time to look into this!

The theme doesn't contribute much, just one of the factors. It's the non-tabularity that's inconvenient. Right now it feels like a spreadsheet with column names repeated above each row.

Current UI:

![](https://i.imgur.com/CRyO6wx.png)

Without headers:

![](https://i.imgur.com/oTgQCHH.png)

A bit more density, the "assert that" on the left of the test strings, and a bit more color coding would be even better, but just the headerless variant would go a long way.

About `description` being pointless, the key part was "for me". As you can see above, the test string itself and the assertions are self-descriptive. A middle ground could be to have `${assertion} ${testString}` in place of an unspecified `description` (i.e. in place of "Unspecified unit test" in the screenshot above).

**Comment by firasdib** [comment]
What about this: if no description is given, we render the compact mode?


Vänliga hälsningar / Best regards,

Firas Dib

**Comment by jeremejevs** [comment]
Sure, that could work!

**Comment by firasdib** [comment]
I've made a broader issue here: https://github.com/firasdib/Regex101/issues/893

Feel free to post your ideas, screenshots and thoughts there.

**Comment by jeremejevs** [comment]
I'll leave the brainstorming to the contributers, I'm just glad to have my usecase taken into consideration.

**Comment by firasdib** [comment]
Voice your concern, and take part of the discussion, or it might be left out!
